### PR TITLE
Try avx2 build again

### PR DESCRIPTION
**Pull request by nickitat**
### Changelog category (leave one):
- Performance Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
todo

---

Continuation of the abandoned efforts here: #36593.

- [x] check perf with `prefer-vector-width` - bad for 128 and 256, because avx512 specialisations degrade
- [x] check perf with `avx2 + prefer-vector-width=128 + allow-light-256-bit`
- [ ] provide specialisations for logical functions

### CODE REVIEW

File: cmake/cpu_features.cmake [hunk 1]
```diff
@@ -224,7 +224,7 @@ elseif (ARCH_AMD64)
         set (COMPILER_FLAGS "${COMPILER_FLAGS} ${TEST_FLAG}")
     endif ()
 
-    set (TEST_FLAG "-mavx2")
+    set (TEST_FLAG "-mavx2 -mprefer-vector-width=128")
```
Comment (nickitat) [inline cmake/cpu_features.cmake:227]:
```suggestion
    set (TEST_FLAG "-mavx2 -mprefer-vector-width=256")
```

File: cmake/cpu_features.cmake [hunk 2]
```diff
@@ -224,7 +224,7 @@ elseif (ARCH_AMD64)
         set (COMPILER_FLAGS "${COMPILER_FLAGS} ${TEST_FLAG}")
     endif ()
 
-    set (TEST_FLAG "-mavx2")
+    set (TEST_FLAG "-mavx2 -mprefer-vector-width=256")
```
Comment (nickitat) [inline cmake/cpu_features.cmake:227]:
```suggestion
    set (TEST_FLAG "-mavx2")
```

**Comment by nickitat** [comment]
what do we have btw on the test machines:
* Skylake 8175M or Cascade Lake 8259CL
* AMD EPYC 7571
* Cascade Lake 8223CL or Skylake 8124M

would be interesting to test ice lake too

**Comment by nickitat** [comment]
from the first glance all slowed down tests are those that use heavy avx like int multiplication and fp.

**Comment by alexey-milovidov** [comment]
Ok.

**Pull request closed without merging**